src/lib/comms/sol-lwm2m-client.c

```diff
@@ -2131,11 +2181,12 @@ server_connection_ctx_new(struct sol_lwm2m_client *client,
     SOL_INT_CHECK_GOTO(r, < 0, err_append);
     conn_ctx->client = client;
     conn_ctx->server_id = server_id;
+    conn_ctx->secure = secure;
     sol_vector_init(&conn_ctx->server_addr_list,
         sizeof(struct sol_network_link_addr));
 
     if (!uri.port)
-        conn_ctx->port = SOL_LWM2M_DEFAULT_SERVER_PORT;
```

Comment:
I guess we should check `conn_ctx->secure` in here. If it `true` we should use the default DTLS port.